api/v1beta1/azurecluster_validation_test.go

```diff
@@ -439,6 +467,32 @@ func TestValidateVnetCIDR(t *testing.T) {
 	}
 }
 
+func TestClusterSubnetsValid(t *testing.T) {
+	type test struct {
+		name    string
+		subnets Subnets
+	}
+
+	testCase := test{
+		name: "subnets - valid",
+		subnets: Subnets{
+			{
+				SubnetClassSpec: SubnetClassSpec{
+					Role: SubnetCluster,
+					Name: "cluster-subnet",
+				},
+			},
+		},
+	}
```

Comment:
Can we add some test cases for invalid? Looks like most tests in this file only test for valid subnets

Reply:
Sure will do that.